### DISCUSSION TITLE
Ask for advice on generalization capabilities of deepmd-kit

### DISCUSSION THREAD

**Discussion by bizhisheng** [deepmodeling/deepmd-kit#2813] [Q&A]
Hello everyone:

I want to ask you a question.
In recent exploration, I found that deepmd seems to be unable to "be compatible with multi-element systems to low-element systems, nor can it perform coupling between different systems." For example, I used deep to fit the potential function graph.pb of the better SiO2-CaO binary system, but graph.pb cannot be applied to pure SiO2 and pure CaO systems. For another example, deep fits the potential function of the SiO2-CaO and SiO2-TiO2 systems better, but it cannot be applied to the SiO2-CaO-TiO2 system.
     Therefore, I would like to ask you whether it is possible to achieve "compatibility between multi-element systems and low-element systems, as well as coupling between different systems." Or, is it possible to give a certain empirical potential function type (such as Morse potential function) and let deepmd correct the relevant parameters.

Thanks for reading!

**Comment by wanghan-iapcm** [comment]
these are typical out-of-distribution generalization, so the accuracy is not guaranteed.  
However, it is expected that you may train SiO2 or CaO by starting from SiO2-CaO by using much less data than training them from scratch.

**Reply by bizhisheng** [reply]
Dear professor of Wang:

Hello, during the recent exploration process, our research team learned about the software **DP-GEN**, in which the exploration-labeling-training process includes the setting of the AIMD initial configuration and reaction conditions. Therefore, I would like to ask whether **DP-GEN** can effectively achieve generalization of the same system composition and temperature.

Thank you for your reading!

**Reply by bizhisheng** [reply]
Hello, Professor Feng:

Recently, the running process of **Deepmd-kit** seems to be not so smooth. I now have three sets of data calculated by **VASP**, namely 1873K, 2273K and 2473K. I want to use these data to fit PES in a wide temperature range, but the result is not ideal, and the **OUTCAR** file processed by dpdata does not contain temperature information.

Professor Feng, I want to ask two questions.
The first is how to make deepmd recognize the training data of 1873K, 2273K and 2473K. There is only one set of data in official website's tutorial, and how to operate this multi-set of data.
The second is whether **dpdata** can export **temperature.npy** and submit it to **Deepmd-kit** for reference.

Looking forward to your reading! I wish you good health!

**Comment by Taixi-Feng** [comment]
Regarding your first example, it's important to note that Deep Potential is developed based on the position and type of atoms. The energy and force of a binary system may differ greatly from that of a pure system. The original PES may not be unavailable  for the pure system.

In your second example, it seems that the forces and energies between CaO and TiO2 in the SiO2-CaO-TiO2 system cannot be described.

If you're looking for a certain empirical potential function type and want to correct the relevant parameters, you may consider using the DMFF package.

**Reply by bizhisheng** [reply]
Dear professor of Feng:

Hello, during the recent exploration process, our research team learned about the software **DP-GEN**, in which the exploration-labeling-training process includes the setting of the AIMD initial configuration and reaction conditions. Therefore, I would like to ask whether **DP-GEN** can effectively achieve generalization of the same system composition and temperature.

Thank you for your reading!

**Reply by Taixi-Feng** [reply]
My recent research results indicate that DP-GEN is feasible in achieving generalization of the same system composition and temperature

**Reply by bizhisheng** [reply]
Dear Professor Feng:

Thank you very much for your reply! After we train the model, we often need to check its quality. You gave an energy example on the official website (https://docs.deepmodeling.com/projects/deepmd/en/master/getting-started/quick_start.html#test-a-model), but there are no force and virial examples. So I would like to ask if I haven’t found the case, or if I need to explore it myself.

Thank you for reading! Good luck with your work!

**Reply by Taixi-Feng** [reply]
**dp test** can provide the RMSE about energy, force and virial.

**Reply by bizhisheng** [reply]
Dear Professor Feng:

Thank you very much for your reply! Sorry, I couldn't express myself clearly. I understand what you mean, you are referring to the relevant information in the **lcurve.out** file. What is more needed in the paper is to correspond the predicted value of the **DPMD model with the calculated value of DFT**, as shown in the example on the official website (https://docs.deepmodeling.com/projects/deepmd/en/master/getting-started/quick_start.html#test-a-model).

Thank you for reading! Good luck with your work!

**Reply by Taixi-Feng** [reply]
Do you want to compare the values of energy, force, and Virial stress predicted by DPMD with DFT

**Reply by bizhisheng** [reply]
Dear Professor Feng:

Yes, can it be achieved?

Thank you for reading! Good luck with your work!

**Reply by Taixi-Feng** [reply]
**dp test** command can achieve this task.

**Reply by bizhisheng** [reply]
Dear Professor Feng:

     Thank you very much for your reply! I have no doubts for now!

Good luck with your work!

**Reply by bizhisheng** [reply]
Dear professor **Feng**

Hello, according to the CH4 case provided by official website, I successfully run **DPGEN**, but when I changed the system to SiO2-TiO2, an error occurred（  **File "/lustre/home/acct-msehqd/msehqd/.local/lib/python3.11/site-packages/dpdata/deepmd/raw.py", line 67, in dump
    nframes = data["cells"].shape[0]
              ^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'shape'**
）. The only difference between CH4 and SiO2-TiO2 cases is that the training data of deepmd are different and the temperature (SiO2-TiO2 temperature is 1873K), and the rest are the same, with no errors in element names and file paths.

Thank you for reading! Looking forward to your reply!
I wish you all the best!

**Reply by Taixi-Feng** [reply]
It seems that there was a problem when generating the datasets using dpdata package.

**Reply by bizhisheng** [reply]
Yes, **Professor Feng.** Error in link 08 of **record.dpgen** file. 06-08 fp link, I used **CP2K**. This error has never been solved. I wonder if **Professor Feng** has ever encountered it and if there is any solution. Exactly the same idea can be applied to CH4.

**Reply by bizhisheng** [reply]
Hello, Professor Feng:

Recently, the running process of **Deepmd-kit** seems to be not so smooth. I now have three sets of data calculated by **VASP**, namely 1873K, 2273K and 2473K. I want to use these data to fit PES in a wide temperature range, but the result is not ideal, and the **OUTCAR** file processed by **dpdata** does not contain temperature information.

Professor Feng, I want to ask two questions.
The first is how to make deepmd recognize the training data of 1873K, 2273K and 2473K. There is only one set of data in official website's tutorial, and how to operate this multi-set of data.
The second is whether **dpdata** can export **temperature.npy** and submit it to **Deepmd-kit** for reference.

Looking forward to your reading! I wish you good health!

**Comment by Taixi-Feng** [comment]
Please provide the files.

---- Replied Message ----
| From | ***@***.***> |
| Date | 09/26/2023 17:08 |
| To | ***@***.***> |
| Cc | cd-feng ***@***.***>,
Comment ***@***.***> |
| Subject | Re: [deepmodeling/deepmd-kit] Ask for advice on generalization capabilities of deepmd-kit (Discussion #2813) |

Yes, Professor Feng. Error in link 08 of record.dpgen file. 06-08 fp link, I used CP2K. This error has never been solved. I wonder if Professor Feng has ever encountered it and if there is any solution. Exactly the same idea can be applied to CH4.

**Reply by bizhisheng** [reply]
[first.zip](https://github.com/deepmodeling/deepmd-kit/files/12725639/first.zip)
Professor Feng, here are all the files.

**Reply by Taixi-Feng** [reply]
DP-GEN repository  #1237
